=> Using adam solver for optimization
=> Using cos for criterion normal
/data2/datasets/ruoguli/miniconda/envs/torch_env/lib/python3.12/site-packages/torch/optim/lr_scheduler.py:224: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn(
/data2/datasets/ruoguli/miniconda/envs/torch_env/lib/python3.12/site-packages/torch/optim/lr_scheduler.py:595: UserWarning: To get the last learning rate computed by the scheduler, please use `get_last_lr()`.
  _warn_get_lr_called_within_step(self)
---- Start val Epoch 1: 33 batches ----
Val:   0%|                                            | 0/33 [00:00<?, ?it/s]/data2/datasets/ruoguli/miniconda/envs/torch_env/lib/python3.12/site-packages/torch/utils/data/dataloader.py:617: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(
/home/ruoguli/idl_project/utils/eval_utils.py:12: UserWarning: indexing with dtype torch.uint8 is now deprecated, please use a dtype torch.bool instead. (Triggered internally at /opt/conda/conda-bld/pytorch_1729647329220/work/aten/src/ATen/native/IndexingUtils.h:27.)
  ang_valid   = angular_map[mask.narrow(1, 0, 1).squeeze(1).byte()]
Traceback (most recent call last):                                           
---------- VAL Epoch 1 Summary -----------
  File "/home/ruoguli/idl_project/train_PSFCN.py", line 51, in <module>
    main(args)
  File "/home/ruoguli/idl_project/train_PSFCN.py", line 45, in main
    val_opt = test_utils.test(args, 'val', val_loader, model, log, epoch, recorder)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ruoguli/idl_project/test_utils.py", line 44, in test
    log.printEpochSummary(opt)
  File "/home/ruoguli/idl_project/utils/logger.py", line 60, in printEpochSummary
    print(opt['recorder'].epochRecToString(split, epoch))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ruoguli/idl_project/utils/recorders.py", line 67, in epochRecToString
    for k in self.records[split].keys():
             ~~~~~~~~~~~~^^^^^^^
KeyError: 'val'
